docs/user-guide/lazy-loading.md

```diff
@@ -0,0 +1,88 @@
+# Lazy Loading
+
+> [!WARNING]
+> This is an experimental option and may not work as expected with all plugins.
+> The API may change without notice. > Please report any issues you encounter.
+
+When the lazy loading functionality is enabled, the plugin's configuration is
+routed to the enabled lazy provider. Options mentioned will be in relation to
+the `${namespace}.${name}.lazyLoad` context.
+
+## Supported Lazy Loading Providers
+
+It is recommended to become familiar with lazy loading strategies supported by
+your provider in their upstream documentation and the corresponding plugin's
+support.
+
+- [Lz.n](https://github.com/nvim-neorocks/lz.n?tab=readme-ov-file#plugin-spec)
```

Comment:
```suggestion
Currently we only support **one** lazy loader:
- [Lz.n](https://github.com/nvim-neorocks/lz.n?tab=readme-ov-file#plugin-spec)
```

Reply:
Also, wondering if this should have links to our docs too, e.g:

```suggestion
- lz.n: [`plugins.lz-n`](/plugins/lz-n/index.md) - [upstream docs](https://github.com/nvim-neorocks/lz.n?tab=readme-ov-file#plugin-spec)
```

Reply:
I can't run docs on my mac, so will have to verify later that path works :P